bpf/bpf_lxc.c

```diff
@@ -346,9 +346,10 @@ static inline int ipv6_l3_from_lxc(struct __sk_buff *skb,
 		if (ret != TC_ACT_OK)
 			return ret;
 
-#ifndef POLICY_EGRESS
 		send_trace_notify(skb, TRACE_TO_HOST, SECLABEL, HOST_ID, 0, HOST_IFINDEX,
 				  forwarding_reason);
+
+#ifndef POLICY_EGRESS
```

Comment:
I think previously these `POLICY_EGRESS` ifdefs avoided calling `send_trace_notify()` because it was expected that the `cilium_reserved_policy` tail call would do the appropriate tracing.

If we want the tracing like this patch has changed it, we should apply the same to IPv4; otherwise we should probably drop these changes.

Reply:
@joestringer I assume you are talking about the following:

``` c
        if (policy_can_access(&POLICY_MAP, skb, src_label, 0, 0, 0, NULL) != TC_ACT_OK) {
                return send_drop_notify(skb, src_label, SECLABEL, 0, ifindex,
                                        DROP_POLICY, TC_ACT_SHOT);
        } else {
                cilium_dbg_capture(skb, DBG_CAPTURE_DELIVERY, ifindex);
```

The main issue I have with the current code is that the `TO_HOST` trace is only emitted if policy is disabled. The policy code will later emit a debug capture but it is subject to debugging being enabled whereas tracing is not.

I will add a debug capture just before the `redirect()` in the policy disabled case so we have symmetry and also emit a debug capture for the policy disabled case.

Regarding IPv6, good catch!